test/modules/diten/test_use_chain_resolution.good

```diff
@@ -1,2 +1,9 @@
+test_use_chain_resolution.chpl:23: In function 'main':
+test_use_chain_resolution.chpl:25: warning: potentially surprising shadowing for 'aaa'
+test_use_chain_resolution.chpl:24: note: it refers to a symbol found through the 'use' statement here
```

Comment:
This test to me illustrates that this warning should mention the module being used/imported to cause the issue.  There are two modules being used on line 24 and you only know which module is involved by looking at the line number on the following line (which will be far less findable in real modules)

Reply:
Does this seem better?

```
test_use_chain_resolution.chpl:23: In function 'main':
test_use_chain_resolution.chpl:25: warning: potentially surprising shadowing for 'aaa'
test_use_chain_resolution.chpl:24: note: it refers to a symbol found through the 'use M3' statement here
test_use_chain_resolution.chpl:10: note: and then through the 'use M2' statement here
test_use_chain_resolution.chpl:6: note: and then through the 'use M1' statement here
test_use_chain_resolution.chpl:2: note: leading to 'aaa' defined here
test_use_chain_resolution.chpl:22: note: but, there is a shadowed symbol 'aaa' defined here
test_use_chain_resolution.chpl:2: error: symbol aaa is multiply defined
test_use_chain_resolution.chpl:14: note: also defined here
```

One concern I have about this approach is that it would print e.g. `use X` even for `use X only Foo` (not for this error but in other places printing the module use tracing -- for this error it might shorten `use X except Bar` to `use X`). Similarly, it would shorten `import X.Y` into `import X`. In this way, it could be misleading.

Anyway with `--detailed-errors`, the relevant portion of the `use` statement is printed out and underlined.

Curious what you think about the tradeoffs here & if you have another suggestion for where to put the module name.

Reply:
I worked on the error wording a bit with @DanilaFe and we came up with this:

```
test_use_chain_resolution.chpl:23: In function 'main':
test_use_chain_resolution.chpl:25: warning: potentially surprising shadowing for 'aaa'
test_use_chain_resolution.chpl:24: note: it refers to a symbol found through the 'use' of 'M3'  here
test_use_chain_resolution.chpl:10: note: and then through the 'use' of 'M2'  here
test_use_chain_resolution.chpl:6: note: and then through the 'use' of 'M1'  here
test_use_chain_resolution.chpl:2: note: leading to 'aaa' defined here
test_use_chain_resolution.chpl:22: note: but, there is a shadowed symbol 'aaa' defined here
test_use_chain_resolution.chpl:2: error: symbol aaa is multiply defined
test_use_chain_resolution.chpl:14: note: also defined here
```

This wording avoids a problem with the redefinition error (which uses the same code to print out how we arrived at something through use/import).

Reply:
I prefer the second error wording and would have suggested exactly that if Daniel hadn't (so good job :) )